runtime/jcl/common/jclexception.c

```diff
@@ -372,3 +374,36 @@ getStackTrace(J9VMThread * vmThread, j9object_t * exceptionAddr, UDATA pruneCons
 
 	return result;
 }
+
+/**
+ * Set the includeClassLoaderName and includeModuleVersion fields for a StackTraceElement.
+ *
+ * @param vmThread The VM thread.
+ * @param element The element to set fields for.
+ * @param classLoader The classloader to check.
+ */
+static void
+setStackTraceElementFields(J9VMThread *vmThread, j9object_t element, J9ClassLoader *classLoader) {
+	J9JavaVM *vm = vmThread->javaVM;
+	BOOLEAN includeClassLoaderName = TRUE;
+	BOOLEAN includeModuleVersion = TRUE;
+
+	/**
+	 * If the classloader is one of the Platform or Bootstrap built-in classloaders,
+	 * don't include its name or module version in the stack trace. If it is the
+	 * Application/System built-in classloader, don't include the class name, but
+	 * include the module version.
+	 */
+	if ((NULL == classLoader)
+		|| (vm->extensionClassLoader == classLoader) // JRE: Platform ClassLoader
+		|| (vm->systemClassLoader == classLoader) // JRE: Bootstrap ClassLoader
+	) {
+		includeClassLoaderName = FALSE;
+		includeModuleVersion = FALSE;
+	} else if (vm->applicationClassLoader == classLoader) { // JRE: System ClassLoader
+		includeClassLoaderName = FALSE;
+	}
+
+	J9VMJAVALANGSTACKTRACEELEMENT_SET_INCLUDECLASSLOADERNAME(vmThread, element, includeClassLoaderName);
```

Comment:
This function and caller need an #ifdef. Not sure if this will compile for Java 8, but it won't work at runtime.

Reply:
The caller is contained within an existing check for `if (j2seVersion >= J2SE_V11)` -- should that code be ifdeffed on JAVA 11 as well?

Reply:
Probably fine then, we'll find out when I try a jdk8 compile.